### ISSUE TITLE
fontbakery --version may return incorrect version number

### ISSUE THREAD

**Issue by m4rc1e** [googlefonts/fontbakery#2198]
`fontbakery --version` will return the version number at the point of installing the package. It won't update if the user has git pulled from the repo.

from the convo in https://github.com/googlefonts/fontbakery/issues/2194

**Comment by felipesanches** [comment]
are you sure about that?

I noticed that Lasse did not run fontbakery --version right after fetching and rebasing. At that point we could have seen a different output, but that possibility was not verified yet since Lasse jumped straight to a new pip install command in that example.

**Comment by felipesanches** [comment]
for the record, we're both talking about this specific comment (https://github.com/googlefonts/fontbakery/issues/2194#issuecomment-438347326) by @graphicore, here.

**Comment by felipesanches** [comment]
you mean that you remember that when you ran those commands the fetching and rebasing were actually unnecessary?

**Comment by m4rc1e** [comment]
Just did this

```
$ fontbakery --version
0.0.16.dev2539+g29aea5ac
$ git reset --hard 7a5a29b624de31a521d5c522f662448a1cdaa8ae # pre 0.6
$ fontbakery --version
0.0.16.dev2539+g29aea5ac 
```

If I was to go back to a previous commit, I'd expect the --version flag to reflect this.

**Comment by graphicore** [comment]
Well, I mean `Current branch master is up to date.` is what you get when there was nothing to do:

```
(venv) commander@ghoul:~/Projekte/fontdev/googlefonts/fontbakery$ git checkout master 
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
(venv) commander@ghoul:~/Projekte/fontdev/googlefonts/fontbakery$ git branch test_branch HEAD~10         
(venv) commander@ghoul:~/Projekte/fontdev/googlefonts/fontbakery$ git checkout test_branch 
Switched to branch 'test_branch'
(venv) commander@ghoul:~/Projekte/fontdev/googlefonts/fontbakery$ git rebase  master 
First, rewinding head to replay your work on top of it...
Fast-forwarded test_branch to master.
(venv) commander@ghoul:~/Projekte/fontdev/googlefonts/fontbakery$ git rebase  master 
Current branch test_branch is up to date.
```

**Comment by anthrotype** [comment]
That’s correct. It is by design like that. Setuptools_scm generates the version string from git only upon running the setup.py script (by writing that _version.py file which is .gitignored).

If you want the version string to be dynamically generated every time one imports fontbakery, you could use versioneer.py

https://github.com/warner/python-versioneer

Personally I prefer setuptools_scm, and don’t mind re-doing pip install -e . after each git pull.
